inference.py

```diff
@@ -225,47 +227,83 @@ def main(params: dict):
+
+    # MANDATORY PARAMETERS
+    img_dir_or_csv = get_key_def('img_dir_or_csv_file', params['inference'], expected_type=str)
+    assert Path(img_dir_or_csv).exists(), f'Couldn\'t locate file or directory "{img_dir_or_csv}" ' \
+                                          f'containing imagery for inference'
+    state_dict = get_key_def('state_dict_path', params['inference'], expected_type=str)
+    assert Path(state_dict).is_file(), f'Couldn\'t locate state_dict of model "{state_dict}" to be used for inference'
+    task = get_key_def('task', params['global'], default='segmentation', expected_type=str)
+    model_name = get_key_def('model_name', params['global'], expected_type=str).lower()
+    num_classes = get_key_def('num_classes', params['global'], expected_type=int)
+    num_bands = get_key_def('number_of_bands', params['global'], expected_type=int)
+
+    # OPTIONAL PARAMETERS
+    chunk_size = get_key_def('chunk_size', params['inference'], default=512, expected_type=int)
+    dontcare_val = get_key_def("ignore_index", params["training"], default=-1, expected_type=int)
+    num_devices = get_key_def('num_gpus', params['global'], default=0, expected_type=int)
+
+    # SETTING OUTPUT DIRECTORY
     working_folder = Path(params['inference']['state_dict_path']).parent.joinpath(f'inference_{num_bands}bands')
-    num_devices = params['global']['num_gpus'] if params['global']['num_gpus'] else 0
     Path.mkdir(working_folder, parents=True, exist_ok=True)
-    print(f'Inferences will be saved to: {working_folder}\n\n')
 
+    # SETUP LOGGING
+    import logging.config  # See: https://docs.python.org/2.4/lib/logging-config-fileformat.html
+    log_config_path = Path('utils/logging.conf').absolute()
+    logfile = f'{working_folder}/info.log'
+    logfile_debug = f'{working_folder}/debug.log'
+    logging.config.fileConfig(log_config_path, defaults={'logfilename': logfile, 'logfilename_debug': logfile_debug})
+    logging.info(f'Inferences will be saved to: {working_folder}\n\n')
+
+    # mlflow logging
+    mlflow_uri = get_key_def('mlflow_uri', params['global'], default=None, expected_type=str)
+    if mlflow_uri:
+        # import only if mlflow uri is set
+        from mlflow import log_params, set_tracking_uri, set_experiment, start_run, log_artifact, log_metrics
+        if not Path(mlflow_uri).is_dir():
+            logging.warning(f"Couldn't locate mlflow uri directory {mlflow_uri}. Directory will be created.")
+            Path(mlflow_uri).mkdir()
+        set_tracking_uri(mlflow_uri)
+        exp_name = get_key_def('mlflow_experiment_name', params['global'], default='gdl-inference', expected_type=str)
+        set_experiment(f'{exp_name}/{working_folder.name}')
+        run_name = get_key_def('mlflow_run_name', params['global'], default='gdl', expected_type=str)
+        start_run(run_name=run_name)
+        log_params(params['global'])
+        log_params(params['inference'])
+
+    # AWS
```

Comment:
all things logging and mlflow in inference should be optional, maybe we can implement a logging flag here or use the existing debug flag (default=false). The usage of inference.py is not limited to only developers.